shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java

```diff
@@ -138,4 +207,140 @@ public String getDescription() {
+        }
+        throw new UnavailableException(
+            ExceptionMessage.BLOCK_UNAVAILABLE.getMessage(info.getBlockId()));
+      }
+      workers = locations.stream().map(BlockLocation::getWorkerAddress).collect(toSet());
+    }
+    // Workers to read the block, after considering failed workers.
+    WorkerNetAddress dataSource = null;
+    locations = locations.stream().filter(location -> workers.contains(location.getWorkerAddress()))
+        .collect(toList());
+    // First try to read data from Alluxio
+    if (!locations.isEmpty()) {
+      // TODO(calvin): Get location via a policy
+      List<WorkerNetAddress> tieredLocations =
+          locations.stream().map(location -> location.getWorkerAddress()).collect(toList());
+      Collections.shuffle(tieredLocations);
+      Optional<Pair<WorkerNetAddress, Boolean>> nearest =
+          BlockLocationUtils.nearest(tieredIdentity, tieredLocations, mFsContext.getClusterConf());
+      if (nearest.isPresent()) {
+        dataSource = nearest.get().getFirst();
+      }
+    }
+
+    // Can't get data from Alluxio, get it from the UFS instead
+    if (dataSource == null) {
+      blockWorkerInfo = blockWorkerInfo.stream()
+          .filter(workerInfo -> workers.contains(workerInfo.getNetAddress())).collect(toList());
+      GetWorkerOptions getWorkerOptions = GetWorkerOptions
+          .defaults().setBlockInfo(new BlockInfo().setBlockId(info.getBlockId())
+              .setLength(info.getLength()).setLocations(locations))
+          .setBlockWorkerInfos(blockWorkerInfo);
+      dataSource = policy.getWorker(getWorkerOptions);
+    }
+    if (dataSource == null) {
+      throw new UnavailableException(ExceptionMessage.NO_WORKER_AVAILABLE.getMessage());
+    }
+    return dataSource;
+  }
+
+  // todo(jianjian) refactoring with InStreamOptions
+  private Protocol.OpenUfsBlockOptions getOpenUfsBlockOptions(long blockId, URIStatus status,
+      OpenFilePOptions options) {
+    Preconditions.checkArgument(status.getBlockIds().contains(blockId), "blockId");
+    boolean readFromUfs = status.isPersisted();
+    // In case it is possible to fallback to read UFS blocks, also fill in the options.
+    boolean storedAsUfsBlock = status.getPersistenceState().equals("TO_BE_PERSISTED");
+    readFromUfs = readFromUfs || storedAsUfsBlock;
+    if (!readFromUfs) {
+      return Protocol.OpenUfsBlockOptions.getDefaultInstance();
+    }
+    BlockInfo info = status.getBlockInfo(blockId);
+    long blockStart = BlockId.getSequenceNumber(blockId) * status.getBlockSizeBytes();
+    Protocol.OpenUfsBlockOptions openUfsBlockOptions = Protocol.OpenUfsBlockOptions.newBuilder()
+        .setUfsPath(status.getUfsPath()).setOffsetInFile(blockStart).setBlockSize(info.getLength())
+        .setMaxUfsReadConcurrency(options.getMaxUfsReadConcurrency())
+        .setNoCache(!ReadType.fromProto(options.getReadType()).isCache())
```

Comment:
This is the problem here.. the  readType was not set in the options, and NoCache is set to true..  which causes the cache function to not cache anything at all. I think it makes sense to setNoCache to false here..

Reply:
solved by creating the default option in the beginning instead.